src/OlStyler.js

```diff
@@ -70,6 +70,52 @@ function hexToRGB(hex, alpha) {
 }
 
 function polygonStyle(style) {
+
+  if (style.fill
+        && style.fill.graphicfill
+        && style.fill.graphicfill.graphic
+        && style.fill.graphicfill.graphic.externalgraphic
+        && style.fill.graphicfill.graphic.externalgraphic.onlineresource) {
+    // Check symbolizer metadata to see if the image has already been loaded.
+    switch (style.__loadingState) {
+      case IMAGE_LOADED:
+        const { image, width, height } = imageCache[style.fill.graphicfill.graphic.externalgraphic.onlineresource]
+        const maxSide = Math.max(width, height)
+
+        let imageRatio = 1
+        if (style.fill.graphicfill.graphic.size && maxSide != style.fill.graphicfill.graphic.size) {
+            imageRatio = style.fill.graphicfill.graphic.size / maxSide
```

Comment:
The SLD spec says that the image should scale with image height, not with max side.
Relevant line from the spec:
> If a size is specified, the height of the graphic will be scaled to that size and the corresponding aspect will be used for the width.

Reply:
okay